src/commands/database/list.mjs

```diff
@@ -1,33 +1,98 @@
 //@ts-check
 
+import { FaunaError } from "fauna";
+
 import { container } from "../../cli.mjs";
-import { commonQueryOptions } from "../../lib/command-helpers.mjs";
+import { throwForError } from "../../lib/fauna.mjs";
 import { FaunaAccountClient } from "../../lib/fauna-account-client.mjs";
 
-async function listDatabases() {
-  const logger = container.resolve("logger");
+// Narrow the output fields based on the provided flags.
+const getOutputFields = (argv) => {
+  if (!argv.secret && !argv.database) {
+    // If we are listing top level databases the region group
+    // needs to be included as database names can be re-used across
+    // regions.
+    return ["name", "region_group"];
+  }
+  return ["name"];
+};
 
-  // query the account api
-  const accountClient = new FaunaAccountClient();
-  const databases = await accountClient.listDatabases();
-  logger.stdout(databases);
-
-  // see what credentials are being used
-  const credentials = container.resolve("credentials");
-  logger.debug(
-    `
-    Account Key: ${credentials.accountKeys.key}\n
-    Database Key: ${credentials.databaseKeys.key}`,
-    "creds",
+function pickOutputFields(databases, argv) {
+  return databases.map((d) =>
+    getOutputFields(argv).reduce((acc, field) => {
+      acc[field] = d[field];
+      return acc;
+    }, {}),
   );
 }
 
+async function listDatabasesWithAccountAPI(argv) {
+  const { pageSize, database, json } = argv;
```

Comment:
why are we not supporting pagination?

Reply:
I didn't want to tackle that here but I can in a follow up